src/Libraries/UnitsUI/UnitsUI.cs

```diff
@@ -381,4 +276,577 @@ public override IEnumerable<AssociativeNode> BuildOutputAst(List<AssociativeNode
+            string value = updateValueParams.PropertyValue;
+
+            if (name == "Value")
+            {
+                Value = value;
+                return true; // UpdateValueCore handled.
+            }
+
+            // There's another 'UpdateValueCore' method in 'String' base class,
+            // since they are both bound to the same property, 'StringInput' 
+            // should be given a chance to handle the property value change first
+            // before the base class 'String'.
+            return base.UpdateValueCore(updateValueParams);
+        }
+
+        protected override void SerializeCore(XmlElement nodeElement, SaveContext context)
+        {
+            base.SerializeCore(nodeElement, context);
+            XmlElement outEl = nodeElement.OwnerDocument.CreateElement(typeof(string).FullName);
+
+            var helper = new XmlElementHelper(outEl);
+            helper.SetAttribute("value", SerializeValue());
+            nodeElement.AppendChild(outEl);
+        }
+
+        protected override void DeserializeCore(XmlElement nodeElement, SaveContext context)
+        {
+            base.DeserializeCore(nodeElement, context);
+            foreach (XmlNode subNode in nodeElement.ChildNodes)
+            {
+                if (subNode.Name.Equals(typeof(string).FullName))
+                {
+                    foreach (XmlAttribute attr in subNode.Attributes)
+                    {
+                        if (attr.Name.Equals("value"))
+                        {
+                            Value = DeserializeValue(attr.Value);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    [NodeCategory(BuiltinNodeCategories.CORE_UNITS)]
+    [NodeName("Convert By Units")]
+    [NodeDescription("ConvertUnitsDescription", typeof(UnitsUI.Properties.Resources))]
+    [NodeSearchTags("ConvertUnitsSearchTags", typeof(UnitsUI.Properties.Resources))]
+    [OutPortTypes("number")]
+    [IsDesignScriptCompatible]
+    public class DynamoUnitConvert : NodeModel
+    {
+        private DynamoUnits.Unit selectedFromConversion;
+        private DynamoUnits.Unit selectedToConversion;
+        private DynamoUnits.Quantity selectedQuantityConversion;
+        private bool isSelectionFromBoxEnabled;
+        private string selectionFromBoxToolTip;
+        private IEnumerable<DynamoUnits.Quantity> quantityConversionSource;
+        private IEnumerable<DynamoUnits.Unit> selectedFromConversionSource;
+        private IEnumerable<DynamoUnits.Unit> selectedToConversionSource;
+        private const string defaultSelectedQuantity = "Length"; 
+
+        /// <summary>
+        /// The NodeType property provides a name which maps to the
+        /// server type for the node. This property should only be
+        /// used for serialization.
+        /// </summary>
+        public override string NodeType
+        {
+            get
+            {
+                return "ConvertUnitsNode";
```

Comment:
I think Extension node yes?

Reply:
Looking at other nodes this is always very specific.  Color Pallete node -> ColorInputNode.  Number Slider -> NumberInputNode

Reply:
but isn't that because those were types defined in the cogs schema? I'll have to check the deserialization code, but I *thought* we didn't deserialize all properties unless the type was ExtensionNode - maybe newtonsoft does it anyway...